k8s-dev/cstor-pool-mgmt/deployment-cstor-pool.yaml

```diff
@@ -0,0 +1,72 @@
+# The deployment for cstor-pool and cstor-pool-mgmt.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cstor-pool
+  labels:
+    app: cstor-pool
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: cstor-pool
+  template:
+    metadata:
+      labels:
+        app: cstor-pool
+    spec:
+      containers:
+      - name: cstor-pool
+#The image name here will be changed to openebs once the business logic pr is reviewed and merged in openebs/maya.
+        image: gkganeshr/cstor-pool:v1.1
+        ports:
+        - containerPort: 80
+        securityContext: 
+          privileged: true
+
+        volumeMounts:
+        - name: device
+          mountPath: /dev     
+        - name: tmp
+          mountPath: /tmp
+#To avoid clash between terminating and restarting pod, we keep initial delay.
+        lifecycle:
+          postStart:
+            exec:
+              command: ["/bin/sh", "-c", "sleep 15"]
+
+      - name:    cstor-pool-mgmt
+#The image name here will be changed to openebs once the business logic pr is reviewed and merged in openebs/maya.
+        image: gkganeshr/cstor-pool-mgmt:v1.8
+        ports:
+        - containerPort: 80
+        securityContext: 
+          privileged: true
+        volumeMounts:
+        - name: device
+          mountPath: /dev
+        - name: tmp
+          mountPath: /tmp
+        env:
+          #cstorid env has UID of cStorPool CR
+        - name: cstorid
+          value: "f774f4dd-68a8-11e8-84c7-54e1ad4a9dd4"
+#To avoid clash between terminating and restarting pod, we keep initial delay.
+        lifecycle:
+          postStart:
+            exec:
+              command: ["/bin/sh", "-c", "sleep 15"]
+
+      volumes:
+      - name: device
+        hostPath:
+          # directory location on host
+          path: /dev
+          # this field is optional
+          type: Directory
+      - name: tmp
+        hostPath:
+#From host, dir called /var/openebs/shared-<uid> is created to avoid clash if two replicas run on same node.
+          path: /var/openebs/shared-a2b
+          type: Directory
```

Comment:
do we have an idea how container's /tmp dir will be regularly cleaned? As the container will be storing crash dumps to this directory, it may grow out of bounds in case of repeating crash very quickly. I quite liked Docker's EmptyDir which is cleaned always when the container shuts down unless the container crashed. However this strategy might not be available in k8s or maybe is too aggressive, just a thought.

Reply:
With emptyDir, data will be lost for every container/pod restart. How can we get this crash data with emptyDir if it is lost for restarts? I think we need a periodic/customized workflow for cleaning up our crash cleanups.

Reply:
We can have one more folder also, apart from /tmp. It is chosen as an optimized way to store simple cachefiles. If we are extending to store much bigger crashdumps, we can have hostPath /var/openebs/shared : /var/openebs/shared along with cachefiles.

Reply:
Got this for docker container size in general, https://bobcares.com/blog/docker-container-size/

Reply:
well, the EmptyDir according to the doc is removed only when the pod is removed, but not when the pod crashes (which is exactly the case when core dump is generated):
```
Note: a Container crashing does NOT remove a Pod from a node, so the data in an emptyDir volume is safe across Container crashes.
```
however it could happen that after the crash, it is restarted, runs successfully and then removed from the node, in which case the directory would be still removed. So maybe not the best option :-/  I think it is ok to leave it as it is now.